lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -275,16 +307,48 @@ def _on_update_status(self, event):
 logger = logging.getLogger(__name__)
 
 
+###########################################################################################
+# These are the basic types of errors:
+# * OpenSearchPluginError: base class for all plugin errors in the plugins code
+# * OpenSearchPluginConfigError: base class for all plugin errors related to config options
 class OpenSearchPluginError(OpenSearchError):
     """Exception thrown when an opensearch plugin is invalid."""
 
+    def __init__(self, msg, only_log=False):
+        super().__init__(msg)
+        self._only_log = only_log
+
+    @property
+    def only_log(self) -> bool:
+        """Returns the only_log property."""
+        return self._only_log
+
+
+class OpenSearchPluginConfigError(OpenSearchPluginError):
+    """Exception thrown when opensearch plugin configuration fails."""
+
+    def __init__(self, msg, affected_configs, only_log=False):
+        super().__init__(msg, only_log)
+        self._affected_configs = affected_configs
+
+    @property
+    def affected_configs(self) -> List[str]:
+        """Returns the missing_configs property."""
+        return self._affected_configs
+
+
+###########################################################################################
+
 
 class OpenSearchPluginRelationClusterNotReadyError(OpenSearchPluginError):
     """Exception thrown when making API calls and cluster is not yet ready.
 
     This exception in specific should be handled by classes managing the relations of plugins.
     """
 
+    def __init__(self, only_log=True):
+        super().__init__("The cluster is not ready yet.", only_log)
```

Comment:
related to the point above

Reply:
This error should be logged only. I noticed in the past few CIs we have been throwing exceptions for this error instead. This is a transitory error that happens at certain stages of the cluster.
It is safe to only log it.